streams/src/test/java/org/apache/kafka/streams/state/internals/ReadonlyPartiallyDeserializedSegmentValueTest.java

```diff
@@ -0,0 +1,220 @@
+                        for (final int index : expectedRecordsIndices) {
+                            final long expectedValidTo = index == 0 ? testCase.nextTimestamp : testCase.records.get(index - 1).timestamp;
+
+                            final SegmentSearchResult result = segmentValue.find(from, to, order, segmentRecordIndex);
+                            assert result != null;
+
+                            final TestRecord expectedRecord = testCase.records.get(index);
+
+                            assertThat(result.index(), equalTo(index));
+                            assertThat(result.value(), equalTo(expectedRecord.value));
+                            assertThat(result.validFrom(), equalTo(expectedRecord.timestamp));
+                            assertThat(result.validTo(), equalTo(expectedValidTo));
+
+                            segmentRecordIndex = order.equals(ResultOrder.ASCENDING) ? index - 1 : index + 1;
+                        }
+                        // verify no results within the time range
+                        if (expectedRecordsIndices.size() == 0) {
+                            assertNull(segmentValue.find(from, to, order, -1));
+                        }
+                    }
+                }
+            }
+        }
+
+        @Test
+        public void shouldGetTimestamps() {
+            final byte[] segmentValue = buildSegmentWithInsertLatest(testCase).serialize();
+
+            assertThat(RocksDBVersionedStoreSegmentValueFormatter.getNextTimestamp(segmentValue), equalTo(testCase.nextTimestamp));
+            assertThat(RocksDBVersionedStoreSegmentValueFormatter.getMinTimestamp(segmentValue), equalTo(testCase.minTimestamp));
+        }
+    }
+
+    private static ReadonlyPartiallyDeserializedSegmentValue buildSegmentWithInsertLatest(final TestCase testCase) {
+        SegmentValue segmentValue = null;
+        for (int recordIdx = testCase.records.size() - 1; recordIdx >= 0; recordIdx--) {
+            final TestRecord record = testCase.records.get(recordIdx);
+            final long validTo = recordIdx == 0 ? testCase.nextTimestamp : testCase.records.get(recordIdx - 1).timestamp;
+
+            if (segmentValue == null) {
+                // initialize
+                if (testCase.records.size() > 1 && record.value == null) {
+                    // when possible, validate that building up a segment starting from the degenerate case is valid as well
+                    segmentValue = RocksDBVersionedStoreSegmentValueFormatter.newSegmentValueWithRecord(null, record.timestamp, record.timestamp);
+                } else {
+                    segmentValue = RocksDBVersionedStoreSegmentValueFormatter.newSegmentValueWithRecord(record.value, record.timestamp, validTo);
+                }
+            } else {
+                // insert latest
+                segmentValue.insertAsLatest(record.timestamp, validTo, record.value);
+            }
+        }
+        assert segmentValue != null;
```

Comment:
`assertThat(..., notNull());` ? (Or could we even drop this line?)

Reply:
> `assertThat(..., notNull());` ? (Or could we even drop this line?)

I replaced it with `assertNotNull(segmentValue);`. The intention is to prevent a warning that we have to suppress it. The next line `segmentValue.serialize()` produces the warning: `Method invocation 'serialize' may produce 'NullPointerException'`.